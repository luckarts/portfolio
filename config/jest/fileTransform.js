const path = require('path');

// This is a custom Jest transformer turning file imports into filenames.
// https://jest-bot.github.io/jest/docs/webpack.html

module.exports = {
  process(src, filename) {
    return `module.exports = ${JSON.stringify(path.basename(filename))};`;
  }
};