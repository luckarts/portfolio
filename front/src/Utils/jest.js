import checkPropTypes from "check-prop-types";
import { createStore } from "redux";
import rootReducer from "store/reducer";

export const checkProps = (component, expectedProps) => {
  // @ts-ignore
  const ErrProps = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );

  return ErrProps;
};
export const testStore = initialState => {
  return createStore(rootReducer, initialState);
};
