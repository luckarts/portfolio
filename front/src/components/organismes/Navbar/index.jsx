import React from 'react';
import PropTypes from 'prop-types';
import { activeTab } from 'store/tabs/actions';
import { connect } from 'react-redux';
import './style.scss';
import { NavLink } from 'react-router-dom';

const propTypes = {
	current_tabs: PropTypes.number,
	activeTab: PropTypes.func,
};
const Navbar = ({ activeTab, isAuthenticated }) => {
	return (
		<nav className="nav">
			<div className="medium-container-flex">
				<NavLink
					className="nav-brand"
					onClick={() => {
						activeTab(0);
					}}
					to="/"
				>
					<img src={'/img/logo8.png'} className="nav-logo" alt="logo" />
					<img src={'/img/LOGO7.png'} className="nav-logoName" alt="logoName" />
				</NavLink>
				<div className={isAuthenticated ? 'nav-main admin' : 'nav-main'}>
					<NavLink
						exact
						to="/"
						onClick={() => {
							activeTab(1);
						}}
						activeClassName="nav-link active"
						className="nav-link"
					>
						Projets
					</NavLink>
					<NavLink to="/luc_bachelerie" activeClassName="nav-link active" className="nav-link">
						Ã€ propos
					</NavLink>
					<NavLink to="/illustrations" activeClassName="nav-link active" className="nav-link">
						Galerie
					</NavLink>
				</div>
			</div>
		</nav>
	);
};

Navbar.propTypes = propTypes;
export const mapStateToProps = (state) => ({
	isAuthenticated: state.user.user,
});
export default connect(mapStateToProps, { activeTab })(Navbar);
