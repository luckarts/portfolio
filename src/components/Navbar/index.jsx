import React from 'react';
import PropTypes from 'prop-types';
import { activeTab } from 'store/tabs/actions';
import { connect } from 'react-redux';
import './style.scss';
import { NavLink } from 'react-router-dom';

const propTypes = {
  current_tabs: PropTypes.number,
  activeTab: PropTypes.func
};
const Navbar = ({ activeTab, isAuthenticated }) => {
  const navLinkStyle =
    'transform scale-1 font-medium opacity-75 text-white transition duration-300 ease-in-out  text-1xl ml-10 hover:text-white hover:opacity-100 hover:scale-125';
  const scrollTop = () => {
    window.scrollTo({ top: 0, behavior: 'smooth' });
  };
  return (
    <nav className=" bg-gray-800  fixed xsl:top-0 w-full z-50 py-4 xs:bottom-0 ">
      <div className="flex max-w-screen-xl justify-between m-auto xsl:px-16  sm:px-8">
        <NavLink
          className="flex relative items-center"
          onClick={() => {
            activeTab(0);
          }}
          to="/"
        >
          <img src={'/img/logo8.png'} className="nav-logo" alt="logo" />
          <img src={'/img/LOGO7.png'} className="nav-logoName" alt="logoName" />
        </NavLink>
        <div className={`flex relative items-center ${isAuthenticated && 'mt-2'}`}>
          <NavLink
            exact
            to="/"
            onClick={() => {
              activeTab(1);
              scrollTop();
            }}
            activeClassName="opacity-100 scale-125 "
            className={`${navLinkStyle} `}
          >
            Projets
          </NavLink>
          <NavLink
            to="/luc_bachelerie"
            activeClassName="opacity-100 scale-125 text-white"
            className={`${navLinkStyle}`}
            onClick={() => {
              scrollTop();
            }}
          >
            Ã€ propos
          </NavLink>

          {/* <NavLink to="/illustrations" activeClassName="nav-link active" className="nav-link">
						Galerie
					</NavLink> */}
        </div>
      </div>
    </nav>
  );
};

Navbar.propTypes = propTypes;
export const mapStateToProps = (state) => ({
  isAuthenticated: state.user.user
});
export default connect(mapStateToProps, { activeTab })(Navbar);
