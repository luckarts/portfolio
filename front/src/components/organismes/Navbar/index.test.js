import React from "react";
import { shallow, mount } from "enzyme";
import Navbar from ".";
import { checkProps } from "Utils/jest";
// refactoring
let wrap, props, id, expectedProps, PropsError;
props = {
  current_tabs: 1,
  activeTab: jest.fn()
};

it("Should render.", () => {
  expect(mount(<Navbar {...props} />)).toMatchSnapshot();
});

describe("Galerie Component", () => {
  id = 1;
  beforeEach(() => {
    props = {
      current_tabs: 1,
      activeTab: jest.fn()
    };
    wrap = shallow(<Navbar {...props} />);
  });
  it("renders correctly", () => {
    expect(wrap).toMatchSnapshot();
  });
  it(" Checking PropTypes no error", () => {
    PropsError = checkProps(Navbar, props);
    expect(PropsError).toBeUndefined();
  });
  it(" Checking PropTypes Should not throw a warning", () => {
    expectedProps = {};
    PropsError = checkProps(Navbar, expectedProps);
    expect(PropsError).toEqual(
      "Failed props type: The props `current_tabs` is marked as required in `Navbar`, but its value is `undefined`."
    );
  });
  describe("Galerie Component", () => {
    beforeEach(() => {
      wrap.find(".active").simulate("click");
    });
    it("renders properly", () => {
      expect(props.activeTab).toHaveBeenCalledWith(id);
    });
  });
});
