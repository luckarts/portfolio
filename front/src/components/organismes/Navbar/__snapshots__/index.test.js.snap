// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Failed proptypes renders correctly 1`] = `ShallowWrapper {}`;

exports[`Galerie Component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Galerie Component renders properly 1`] = `ShallowWrapper {}`;

exports[`Should render. 1`] = `ReactWrapper {}`;
